src/main/java/org/openrewrite/java/migrate/AddMissingMethodImplementation.java

```diff
@@ -0,0 +1,148 @@
+
+import com.fasterxml.jackson.annotation.JsonCreator;
+import com.fasterxml.jackson.annotation.JsonProperty;
+import lombok.EqualsAndHashCode;
+import lombok.Value;
+import org.openrewrite.*;
+import org.openrewrite.internal.lang.NonNull;
+import org.openrewrite.java.JavaIsoVisitor;
+import org.openrewrite.java.JavaTemplate;
+import org.openrewrite.java.MethodMatcher;
+import org.openrewrite.java.search.UsesType;
+import org.openrewrite.java.tree.J;
+import org.openrewrite.java.tree.JavaType;
+import org.openrewrite.java.tree.TypeTree;
+
+import java.util.List;
+
+@Value
+@EqualsAndHashCode(callSuper = false)
+public class AddMissingMethodImplementation extends Recipe {
+    @Option(displayName = "Fully Qualified Class Name",
+            description = "A fully qualified class being implemented with missing method.",
+            example = "com.yourorg.FooBar")
+    @NonNull
+    String fullyQualifiedClassName;
+
+    @Option(displayName = "Method Pattern",
+            description = "A method pattern for matching required method definition.",
+            example = "*..* hello(..)")
+    @NonNull
+    String methodPattern;
+
+    @Option(displayName = "Method Template",
+            description = "Template of method to add",
+            example = "public String hello() { return \\\"Hello from #{}!\\\"; }")
+    @NonNull
+    String methodTemplateString;
+
+    // All recipes must be serializable. This is verified by RewriteTest.rewriteRun() in your tests.
+    @JsonCreator
+    public AddMissingMethodImplementation(@NonNull @JsonProperty("fullyQualifiedClassName") String fullyQualifiedClassName,
+                                          @NonNull @JsonProperty("methodPattern") String methodPattern,
+                                          @NonNull @JsonProperty("methodTemplateString") String methodTemplateString) {
+        this.fullyQualifiedClassName = fullyQualifiedClassName;
+        this.methodPattern = methodPattern;
+        this.methodTemplateString = methodTemplateString;
+    }
+
+    @Override
+    public String getDisplayName() {
+        return "Adds missing method implementations.";
+    }
+
+    @Override
+    public String getDescription() {
+        return "Check for missing methods required by interfaces and adds them.";
+    }
+
+    @Override
+    public TreeVisitor<?, ExecutionContext> getVisitor() {
+        return Preconditions.check(new UsesType<>(fullyQualifiedClassName, true), new ClassImplementationVisitor());
+    }
+
+    public class ClassImplementationVisitor extends JavaIsoVisitor<ExecutionContext> {
+        private final JavaTemplate methodTemplate = JavaTemplate.builder(methodTemplateString).build();
+        private final MethodMatcher methodMatcher = new MethodMatcher(methodPattern, true);
+
+        public boolean matchesInterface(JavaType.Class type) {
+            if (type != null) {
+                if (type.getFullyQualifiedName().equals(fullyQualifiedClassName)) {
```

Comment:
I think `TypeUtils` has better methods to check types rather than string comparison. Could we switch to that?

Reply:
I'm not familiar with `TypeUtils` what method did you have in mind?

Reply:
`TypeUtils.isAssignableTo`, as applied in https://github.com/openrewrite/rewrite-migrate-java/pull/312/commits/99833e569ff7b862b2c5505ee265056cde32b352
Quite a bit simpler, for what I think is the same output.